source/php/Component/Icon/Icon.php

```diff
@@ -91,6 +108,74 @@ private function createIconModifier($icon)
         );
     }
 
+    private function getCustomIconPath($path, $icon)
+    {
+        if (!empty(self::$iconsCache[$this->customIconsContentId][$icon])) {
+            return self::$iconsCache[$this->customIconsContentId][$icon];
+        }
+
+        $useWpCache = function_exists('wp_cache_get') && function_exists('wp_cache_set');
+
+        if ($useWpCache) {
+            $cachedIconContent = wp_cache_get($icon, 'customIconsContent');
+            if ($cachedIconContent !== false) {
+                self::$iconsCache[$this->customIconsContentId][$icon] = $cachedIconContent;
+                return $cachedIconContent;
+            }
+        }
+        
+        if (file_exists($path) && is_readable($path)) {
+            $contents = file_get_contents($path);
+            self::$iconsCache[$this->customIconsContentId][$icon] = $contents;
+        }
+
+        if ($useWpCache) {
+            wp_cache_set($icon, self::$iconsCache[$this->customIconsContentId][$icon], 'customIconsContent');
+        }
+
+        return self::$iconsCache[$this->customIconsContentId][$icon] ?? null;
+    }
+
+    private function getCustomSvgIcons() {
+        if (!empty(self::$iconsCache[$this->customIconsId])) {
+            return self::$iconsCache[$this->customIconsId];
+        }
+
+        $useWpCache = function_exists('wp_cache_get') && function_exists('wp_cache_set');
+
+        if ($useWpCache) {
+            $cachedIconList = wp_cache_get($this->customIconsId, 'customIcons');
+            if ($cachedIconList !== false) {
+                self::$iconsCache[$this->customIconsId] = $cachedIconList;
+                return $cachedIconList;
+            }
+        }
+  
+        if (function_exists('apply_filters')) {
+            $svgIcons = apply_filters(
+                'ComponentLibrary\Component\Icon\CustomSvgIcons',
+                glob(__DIR__ . '/Svg/*.svg')
+            );
+        } else {
+            $svgIcons = glob(__DIR__ . '/Svg/*.svg');
+        }
+
+        if (empty($svgIcons)) {
+            return [];
+        }
+
+        foreach ($svgIcons as $svgIcon) {
+            $iconName = pathinfo($svgIcon, PATHINFO_FILENAME);
+            self::$iconsCache['icons'][$iconName] = $svgIcon;
+        }
+    
+        if ($useWpCache) {
```

Comment:
WpCache could be a dependency injection based on a cache interface here. We have used the cche interface twice now, might aswell do it right.

Reply:
Using a common method here, would mean that we get rid of most of the implementation in this class.